src/main/scala/io/chrisdavenport/linebacker/DualContext.scala

```diff
@@ -0,0 +1,43 @@
+package io.chrisdavenport.linebacker
+
+import cats._
+import cats.implicits._
+import cats.effect._
+import fs2.Stream
+import scala.concurrent.ExecutionContext
+
+trait DualContext[F[_]] {
+  def blockingContext: F[ExecutionContext]
+  def defaultContext: F[ExecutionContext]
+
+  def block[A](fa: F[A])(implicit F: Async[F]): F[A] =
+    for {
+      bEC <- blockingContext
+      dEC <- defaultContext
+      _ <- Async.shift(bEC)
+      aE <- fa.attempt
+      _ <- Async.shift(dEC)
+      a <- Applicative[F].pure(aE).rethrow
```

Comment:
You can just write it as F.pure since it inherits.

Reply:
I can but I like the clarity of what capability we're using, and I don't believe this will cause any delay anywhere besides very minimal in compilation, since it needs to hop to the declaration